activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -217,6 +218,10 @@ def connection_retries
         (@config[:connection_retries] || 1).to_i
       end
 
+      def verify_timeout
+        (@config[:verify_timeout] || 5).to_i
```

Comment:
Dropped to 2 seconds. I don't want to go super low, because I don't want e.g. alternating [blocking/waiting] querying of two databases (or even just a "query, do non-trivial CPU work, query again" pattern) to cause repeated verifications within a single request.

Reply:
I've left the hash lookup for now... the fact we're already doing the same thing for `connection_retries` isn't a _good_ argument for that choice, but still I prefer to keep the symmetry. (Also a little reluctant to load up on config-sourced ivars in an already-overwhelmingly-large [and highly stateful] object, but I either way I think we can look at that in a `main`-only follow up.) 

It's technically wasted work (I think the `@config` ivar access is itself probably a hash lookup here, even?), but I'm comfortable that it's sufficiently trivial, relative to the work + latency of the actual querying we're about to do, that it's not an immediate worry.

Reply:
> (I think the `@config` ivar access is itself probably a hash lookup here, even?)

No, it should be a simple pointer offset, as the shape is supposed to be stable.